<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-09-11">

<title>hambiBottomUp data analysis - Calibrate boilprep to Qiagen DNeasy</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../R/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">Community composition amplicon</a></li><li class="breadcrumb-item"><a href="../../../R/amplicon/20240711_BTK_illumina_v3/02_correct_boil_to_qiagen.html">2. 20240711 BTK Illumina v3 / Metacal calibration</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../">hambiBottomUp data analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Experiment design - 96 Well plate layout</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/experiment_design/01_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Species pairs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/experiment_design/02_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Species trios</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/experiment_design/03_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Species quartets</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. 20240711 BTK Illumina v3 / Data wrangling and QC</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/amplicon/20240711_BTK_illumina_v3/02_correct_boil_to_qiagen.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. 20240711 BTK Illumina v3 / Metacal calibration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/amplicon/analysis_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Species pairs analysis</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup"><span class="header-section-number">2</span> Setup</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">2.1</span> Libraries</a></li>
  <li><a href="#global-variables" id="toc-global-variables" class="nav-link" data-scroll-target="#global-variables"><span class="header-section-number">2.2</span> Global variables</a></li>
  </ul></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">3</span> Read data</a></li>
  <li><a href="#formatting" id="toc-formatting" class="nav-link" data-scroll-target="#formatting"><span class="header-section-number">4</span> Formatting</a>
  <ul class="collapse">
  <li><a href="#format-positive-control-samples" id="toc-format-positive-control-samples" class="nav-link" data-scroll-target="#format-positive-control-samples"><span class="header-section-number">4.1</span> Format positive control samples</a></li>
  <li><a href="#plot-masterplate-qiagen-and-boil-prep-side-by-side" id="toc-plot-masterplate-qiagen-and-boil-prep-side-by-side" class="nav-link" data-scroll-target="#plot-masterplate-qiagen-and-boil-prep-side-by-side"><span class="header-section-number">4.2</span> Plot masterplate Qiagen and boil-prep side by side</a></li>
  <li><a href="#matrix-of-observed-counts" id="toc-matrix-of-observed-counts" class="nav-link" data-scroll-target="#matrix-of-observed-counts"><span class="header-section-number">4.3</span> Matrix of observed counts</a></li>
  <li><a href="#make-a-matrix-of-true-proportions" id="toc-make-a-matrix-of-true-proportions" class="nav-link" data-scroll-target="#make-a-matrix-of-true-proportions"><span class="header-section-number">4.4</span> Make a matrix of true proportions</a></li>
  </ul></li>
  <li><a href="#metacal-procedure" id="toc-metacal-procedure" class="nav-link" data-scroll-target="#metacal-procedure"><span class="header-section-number">5</span> Metacal procedure</a>
  <ul class="collapse">
  <li><a href="#estimate-bias" id="toc-estimate-bias" class="nav-link" data-scroll-target="#estimate-bias"><span class="header-section-number">5.1</span> Estimate bias</a>
  <ul class="collapse">
  <li><a href="#plot-bias-estimation" id="toc-plot-bias-estimation" class="nav-link" data-scroll-target="#plot-bias-estimation"><span class="header-section-number">5.1.1</span> Plot bias estimation</a></li>
  <li><a href="#plot-metacal-model-fit" id="toc-plot-metacal-model-fit" class="nav-link" data-scroll-target="#plot-metacal-model-fit"><span class="header-section-number">5.1.2</span> Plot metacal model fit</a></li>
  </ul></li>
  <li><a href="#calibrate" id="toc-calibrate" class="nav-link" data-scroll-target="#calibrate"><span class="header-section-number">5.2</span> Calibrate</a></li>
  <li><a href="#run-the-calibrate-function" id="toc-run-the-calibrate-function" class="nav-link" data-scroll-target="#run-the-calibrate-function"><span class="header-section-number">5.3</span> Run the calibrate function</a></li>
  </ul></li>
  <li><a href="#export" id="toc-export" class="nav-link" data-scroll-target="#export"><span class="header-section-number">6</span> Export</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../R/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">Community composition amplicon</a></li><li class="breadcrumb-item"><a href="../../../R/amplicon/20240711_BTK_illumina_v3/02_correct_boil_to_qiagen.html">2. 20240711 BTK Illumina v3 / Metacal calibration</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Calibrate boilprep to Qiagen DNeasy</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 11, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>We will use the <a href="https://mikemc.github.io/metacal/">metacal package</a> for estimating bias and performing calibration in the special case where the bias of all the taxa of interest can be directly measured from the control sample. Since samples extracted by Qiagen and by boil are exactly the same we can estimate scaling factors to produce corrected relative abundance for other samples.</p>
<p>Publication: <a href="https://elifesciences.org/articles/46923">Consistent and correctable bias in metagenomic sequencing experiments</a></p>
</section>
<section id="setup" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Setup</h1>
<section id="libraries" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">2.1</span> Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(metacal)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(withr)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="global-variables" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">2.2</span> Global variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="read-data" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Read data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>pos_ctrl <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pos_neg_ctrl_counts.tsv"</span>))</span>
<span id="cb3-2"><a href="#cb3-2"></a>samp_pairs <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span>
<span id="cb3-3"><a href="#cb3-3"></a>samp_trios <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co"># metadata</span></span>
<span id="cb3-6"><a href="#cb3-6"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240711_BTK_illumina_v3_metadata.tsv"</span>))</span>
<span id="cb3-7"><a href="#cb3-7"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="formatting" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Formatting</h1>
<section id="format-positive-control-samples" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="format-positive-control-samples"><span class="header-section-number">4.1</span> Format positive control samples</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>pos_ctrl <span class="ot">&lt;-</span> <span class="fu">left_join</span>(pos_ctrl, spdf) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="co"># remove noise from species that aren't really there</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="fu">mutate</span>(<span class="at">f_qg =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-7"><a href="#cb4-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, community_id, count_correct, f_qg) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(strainID, community_id)`</code></pre>
</div>
</div>
</section>
<section id="plot-masterplate-qiagen-and-boil-prep-side-by-side" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="plot-masterplate-qiagen-and-boil-prep-side-by-side"><span class="header-section-number">4.2</span> Plot masterplate Qiagen and boil-prep side by side</h2>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">left_join</span>(samp_trios, mddf,</span>
<span id="cb6-2"><a href="#cb6-2"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="fu">right_join</span>(pos_ctrl, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, community_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, f, f_qg) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(f, f_qg)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb6-8"><a href="#cb6-8"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x=</span><span class="fu">interaction</span>(name, community_id), <span class="at">fill =</span> strainID), </span>
<span id="cb6-9"><a href="#cb6-9"></a>             <span class="at">color=</span><span class="st">"black"</span>,</span>
<span id="cb6-10"><a href="#cb6-10"></a>             <span class="at">linewidth=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb6-11"><a href="#cb6-11"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb6-12"><a href="#cb6-12"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb6-13"><a href="#cb6-13"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"% abundance"</span>) <span class="sc">+</span></span>
<span id="cb6-14"><a href="#cb6-14"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb6-15"><a href="#cb6-15"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-16"><a href="#cb6-16"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb6-17"><a href="#cb6-17"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_correct_boil_to_qiagen_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Bar plot of masterplate samples extracted by the boilprep method (f) or the Qiagen DNeasy kit (qg). The community trios are appended to the extraction method on the x axs
</figcaption>
</figure>
</div>
</section>
<section id="matrix-of-observed-counts" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="matrix-of-observed-counts"><span class="header-section-number">4.3</span> Matrix of observed counts</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>samp_trios_counts <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_trios, mddf) <span class="sc">%&gt;%</span> </span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="fu">filter</span>(community_id <span class="sc">%in%</span> <span class="fu">pull</span>(pos_ctrl, community_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, count_correct) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"community_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-7"><a href="#cb7-7"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(sample, replicate, strep_conc, transfers, n_species,
community_type, plate_well)`</code></pre>
</div>
</div>
</section>
<section id="make-a-matrix-of-true-proportions" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="make-a-matrix-of-true-proportions"><span class="header-section-number">4.4</span> Make a matrix of true proportions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>pos_ctrl_proportions <span class="ot">&lt;-</span> pos_ctrl <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, f_qg) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"f_qg"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"community_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="metacal-procedure" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Metacal procedure</h1>
<section id="estimate-bias" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="estimate-bias"><span class="header-section-number">5.1</span> Estimate bias</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">set.seed</span>(<span class="dv">12378</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a>mc_fit_trios <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">estimate_bias</span>(samp_trios_counts, pos_ctrl_proportions, <span class="dv">1</span>, <span class="at">boot=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-3"><a href="#cb10-3"></a>mc_fit_trios_summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(mc_fit_trios)</span>
<span id="cb10-4"><a href="#cb10-4"></a>mc_fit_trios_summary_coef <span class="ot">&lt;-</span>mc_fit_trios_summary[[<span class="st">'coefficients'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="plot-bias-estimation" class="level3" data-number="5.1.1">
<h3 data-number="5.1.1" class="anchored" data-anchor-id="plot-bias-estimation"><span class="header-section-number">5.1.1</span> Plot bias estimation</h3>
<div id="fig-02" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>mc_fit_trios_summary_coef <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">mutate</span>(<span class="at">taxon =</span> <span class="fu">fct_reorder</span>(taxon, estimate)) <span class="sc">|&gt;</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(taxon, estimate, </span>
<span id="cb11-4"><a href="#cb11-4"></a>             <span class="at">ymin =</span> estimate <span class="sc">/</span> gm_se<span class="sc">^</span><span class="dv">2</span>, <span class="at">ymax =</span> estimate <span class="sc">*</span> gm_se<span class="sc">^</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">color =</span> taxon)) <span class="sc">+</span></span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb11-8"><a href="#cb11-8"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Bias estimate"</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb11-9"><a href="#cb11-9"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb11-10"><a href="#cb11-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb11-11"><a href="#cb11-11"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-12"><a href="#cb11-12"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_correct_boil_to_qiagen_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Bias estimates for each species from the metacal procedure.
</figcaption>
</figure>
</div>
</section>
<section id="plot-metacal-model-fit" class="level3" data-number="5.1.2">
<h3 data-number="5.1.2" class="anchored" data-anchor-id="plot-metacal-model-fit"><span class="header-section-number">5.1.2</span> Plot metacal model fit</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>a <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">fitted</span>(mc_fit_trios)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="st">"Fitted"</span>)</span>
<span id="cb12-5"><a href="#cb12-5"></a></span>
<span id="cb12-6"><a href="#cb12-6"></a>b <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pos_ctrl_proportions) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="st">"Actual"</span>)</span>
<span id="cb12-10"><a href="#cb12-10"></a></span>
<span id="cb12-11"><a href="#cb12-11"></a>c <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samp_trios_counts<span class="sc">/</span><span class="fu">rowSums</span>(samp_trios_counts)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-12"><a href="#cb12-12"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-13"><a href="#cb12-13"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample,  <span class="at">values_to=</span><span class="st">"observed"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="fig-03" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">bind_rows</span>(a,b) <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="fu">left_join</span>(c) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>observed, <span class="at">y=</span>value, <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">geom_abline</span>(<span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb13-7"><a href="#cb13-7"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species proportion in boil-prepped samples"</span>, <span class="at">y =</span> <span class="st">"Species proportion in Qiagen-prepped samples"</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb13-8"><a href="#cb13-8"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>type) <span class="sc">+</span> </span>
<span id="cb13-9"><a href="#cb13-9"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb13-10"><a href="#cb13-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb13-11"><a href="#cb13-11"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-12"><a href="#cb13-12"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(sample, name)`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_correct_boil_to_qiagen_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Porportion of each species from boil-prepped samples (x-axis) and from the Qiagen DNeasy extracted samples (y-axis) colored by species identity. The left panel shows the relationship between the two extraction procedures before correction with metacal. The right panel shows the relationship after correction. The dashed line is 1:1.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="calibrate" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="calibrate"><span class="header-section-number">5.2</span> Calibrate</h2>
<p>Make a matrix of observed counts</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>trios_2_cal <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_trios, mddf) <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(sample, replicate, strep_conc, transfers, n_species,
community_type, plate_well)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>pairs_2_cal <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_pairs, mddf) <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(sample, replicate, strep_conc, transfers, n_species,
community_type, plate_well)`</code></pre>
</div>
</div>
</section>
<section id="run-the-calibrate-function" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="run-the-calibrate-function"><span class="header-section-number">5.3</span> Run the calibrate function</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="fu">set.seed</span>(<span class="dv">435761</span>)</span>
<span id="cb19-2"><a href="#cb19-2"></a></span>
<span id="cb19-3"><a href="#cb19-3"></a>pairs_calibrated <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">calibrate</span>(pairs_2_cal, <span class="fu">coef</span>(mc_fit_trios), <span class="at">margin=</span><span class="dv">1</span>)</span>
<span id="cb19-4"><a href="#cb19-4"></a>trios_calibrated <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">calibrate</span>(trios_2_cal, <span class="fu">coef</span>(mc_fit_trios), <span class="at">margin=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>pairs_calibrated_l <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pairs_calibrated) <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">"strainID"</span>, <span class="at">values_to =</span> <span class="st">"f_metacal"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="fu">filter</span>(f_metacal <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb20-5"><a href="#cb20-5"></a></span>
<span id="cb20-6"><a href="#cb20-6"></a>trios_calibrated_l <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(trios_calibrated) <span class="sc">%&gt;%</span> </span>
<span id="cb20-7"><a href="#cb20-7"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-8"><a href="#cb20-8"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">"strainID"</span>, <span class="at">values_to =</span> <span class="st">"f_metacal"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-9"><a href="#cb20-9"></a>  <span class="fu">filter</span>(f_metacal <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="export" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Export</h1>
<p>Save metacal esimates for later use</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="fu">write_tsv</span>(pairs_calibrated_l, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_metacal.tsv"</span>))</span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="fu">write_tsv</span>(trios_calibrated_l, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_metacal.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>